scripts/pdf_utils.py

```
@@ -58,13 +62,13 @@ def get_paddle_engine():
                         rec_model_dir=str(rec_dir)
                     )
                 except MemoryError:
-                    print("[AI] CRITICAL: Out of Memory while loading PaddleOCR.")
+                    logger.critical("Out of Memory while loading PaddleOCR.")
                     raise MemoryError("Server ran out of memory loading the AI engine. Please upgrade to a 'Starter' plan on Render for this feature.")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)